src/components/Map/Map.jsx

```diff
@@ -5,18 +5,28 @@ import "leaflet/dist/leaflet.css";
 import "leaflet-defaulticon-compatibility/dist/leaflet-defaulticon-compatibility.webpack.css";
 import "leaflet-defaulticon-compatibility";
 
-import { useRef, useEffect } from "react";
+import {useRef, useEffect, useState} from "react";
 import { MapContainer, Marker, Popup, TileLayer } from "react-leaflet";
 import { SelectionOverlay } from "@/components/Map/SelectionOverlay";
 import L from "leaflet";
 import { ItemMarker } from "./ItemMarker";
 import { renderToString } from "react-dom/server";
+import {useChatMutation} from "@/store/services/chatApi";
 
 const Map = (props) => {
   const { position, placesData: places } = props;
-
+  const icon = L.icon({iconUrl: "/images/marker-icon.png"});
+  const [preferences, setPreferences] = useState([]);
   const mapRef = useRef();
+  const [chat, { isLoading, isSuccess, data, error }] = useChatMutation();
```

Comment:
This Chat mutation calls the wrong endpoint. You could use the page.jsx `fetchPlaces` function that is passed to the Map component to fetch based on preference string.

Also, why not use this hook in the component that uses this hook, why pass the fetch function as a prop, when we want to render everything and if it takes time to fetch the preferences, we could use the states to render later the preferences buttons.